src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -633,6 +633,30 @@ public static final boolean indexExists(final Directory directory) throws IOExce
     }
 
     /**
+     * Wait for an index to exist for up to {@code timeLimitMillis}. Returns
+     * true if the index eventually exists, false if not.
+     *
+     * Will retry the directory every second for at least {@code timeLimitMillis}
+     */
+    public static final boolean waitForIndex(final Directory directory, final long timeLimitMillis)
```

Comment:
I think we should just catch InterruptedException here set the interrupted flag and return false.